rotkehlchen/tests/utils/history_base_entry.py

```diff
@@ -0,0 +1,147 @@
+
+from typing import TYPE_CHECKING, Any
+
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.base import HistoryBaseEntry, HistoryEvent
+from rotkehlchen.accounting.structures.evm_event import EvmEvent
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ONE
+from rotkehlchen.constants.assets import A_DAI, A_ETH, A_ETH2, A_USDT
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.api import api_url_for, assert_proper_response_with_result
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+if TYPE_CHECKING:
+    from rotkehlchen.api.server import APIServer
+
+
+def entry_to_input_dict(
+        entry: 'EvmEvent',
+        include_identifier: bool,
+) -> dict[str, Any]:
+    serialized = entry.serialize()
+    if include_identifier:
+        assert entry.identifier is not None
+        serialized['identifier'] = entry.identifier
+    else:
+        serialized.pop('identifier')  # there is `identifier`: `None` which we have to remove
+    serialized.pop('event_identifier')
+    serialized.pop('entry_type')
+    return serialized
+
+
+def add_entries(server: 'APIServer', events_db: DBHistoryEvents, add_directly: bool = False) -> list['HistoryBaseEntry']:  # noqa: E501
+    entries = [EvmEvent(
+        tx_hash=deserialize_evm_tx_hash('0x64f1982504ab714037467fdd45d3ecf5a6356361403fc97dd325101d8c038c4e'),  # noqa: E501
```

Comment:
add a docstring explaining what this function is doing

Reply:
done